### DISCUSSION TITLE
ravendb-jvm-client 5.3 support?

### DISCUSSION THREAD

**Discussion by vijayakumarsuraj** [ravendb/ravendb#13768] [Q&A] [answered]
I was wondering when some of the new features from 5.3 will be made available on the JVM client?
For me, specifically concurrent data subscriptions and exploration queries.

**Comment by ayende** [comment, marked as answer]
We don't have a firm date for that, I'm afraid.

For concurrent subscriptions, you should be able to add `SubscriptionOpeningStrategy.CONCURRENT`, with the rest working as usual.

For exploration query, use `rawQuery` to pass the string directly.

**Reply by vijayakumarsuraj** [reply]
Thanks, @ayende. 

> For concurrent subscriptions, you should be able to add SubscriptionOpeningStrategy.CONCURRENT, with the rest working as usual.

`CONCURRENT` isn't present on the `SubscriptionOpeningStrategy` enum right now. I suppose I could hook into the JSON serialization to _hack_ it in... Or did you mean something else?


> For exploration query, use rawQuery to pass the string directly.

Yep, this is what I'm doing for now. Though, unfortunately, can't write tests against the embedded server with it - since that is 5.2 too.

**Reply by ayende** [reply]
Yes, I meant hack it for now.

As for the tests, you are correct. There was the issue you opened here https://github.com/ravendb/ravendb-jvm-embedded/issues/2 which should help in that regard.

These are hacks, I'm aware. We'll be releasing 5.3 version of JVM, but it is not in the next few weeks, so those should tide you over in the meantime.
